### PR TITLE
centos: add "vault" binary to the image

### PR DESCRIPTION
**Pull request by leseb**
Rook needs the vault binary to interact with a vault server. We cannot
add this to the Rook image since this is executed on Deployments
ressources which run on the host where the OSDs are running. This is not
a command we can run from the Operator itself so it should be in the
Ceph image.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Review by guits — APPROVED** [review]

**Comment by guits** [comment]
@leseb does it need to be backported?

**Comment by guits** [comment]
CI passing, however, I'd like @dsavineau to review it before merging

File: ceph-releases/ALL/centos/daemon-base/__DOCKERFILE_INSTALL__ [hunk 1]
```diff
@@ -45,6 +45,9 @@ bash -c ' \
     else \
       curl -s -L https://download.ceph.com/ceph-iscsi/2/rpm/el__ENV_[BASEOS_TAG]__/ceph-iscsi.repo -o /etc/yum.repos.d/ceph-iscsi.repo ; \
     fi ; \
+  fi ; \
+  if [ -n "__VAULT_PACKAGES__" ] && [[ $(arch) == "x86_64" ]]; then \
```
Comment (dsavineau) [inline ceph-releases/ALL/centos/daemon-base/__DOCKERFILE_INSTALL__:49]: arm64 rpm package isn't available but the precompiled binary is available for that architecture [1]

Is it worth to have both architectures supported for this ? If yes, we need to change the install method.

[1] https://releases.hashicorp.com/vault/1.7.2/vault_1.7.2_linux_arm64.zip

**Review by dsavineau — CHANGES_REQUESTED** [review]
We need to add an empty `__VAULT_PACKAGES__` file for ubi8 other this will break the RHCS container compose

```console
$ ./contrib/compose-rhcs.sh 

  CEPH_VERSION      : pacific
  CEPH_REF          : pacific
  CEPH_POINT_RELEASE: 
  CEPH_DEVEL        : false
  OSD_FLAVOR        : default
  DISTRO            : ubi8
  DISTRO_VERSION    : latest
  HOST_ARCH         : x86_64
  BASEOS_REGISTRY   : registry.redhat.io
  BASEOS_REPO       : ubi8/ubi
  BASEOS_TAG        : latest
  IMAGES_TO_BUILD   : daemon-base daemon
  STAGING_DIR       : staging/pacific-ubi8-latest-x86_64
  RELEASE           : HEAD
  DAEMON_BASE_IMAGE : ceph/daemon-base:HEAD-pacific-ubi8/ubi-latest-x86_64
  DAEMON_IMAGE      : ceph/daemon:HEAD-pacific-ubi8/ubi-latest-x86_64

Computed:
  BASE_IMAGE        : registry.redhat.io/ubi8/ubi:latest

Traceback (most recent call last):
  File "maint-lib/stage.py", line 117, in <module>
    main(CORE_FILES_DIR, CEPH_RELEASES_DIR)
  File "maint-lib/stage.py", line 108, in main
    do_variable_replace(replace_root_dir=os.path.join(STAGING_DIR, image))
  File "maint-lib/stagelib/replace.py", line 131, in do_variable_replace
    total_replacements, rendered_text = _do_replace_on_file(file_path, replace_root_dir)
  File "maint-lib/stagelib/replace.py", line 93, in _do_replace_on_file
    rendered_text, file_path, variable_file_dir=replace_root_dir)
  File "maint-lib/stagelib/replace.py", line 61, in _file_replace
    template_text, file_path)
  File "maint-lib/stagelib/replace.py", line 35, in _replace_file_in_text
    with open(variable_file_path) as variable_file:
FileNotFoundError: [Errno 2] No such file or directory: 'staging/pacific-ubi8-latest-x86_64/daemon-base/__VAULT_PACKAGES__'
make: *** [Makefile:63: stage.x86_64,pacific,ubi8,latest] Error 1
FATAL ERROR !
########################################################
Cannot build rhel8
########################################################
```

File: ceph-releases/ALL/centos/daemon-base/__DOCKERFILE_INSTALL__ (hunk 1 above)
Comment (leseb) [reply ceph-releases/ALL/centos/daemon-base/__DOCKERFILE_INSTALL__:49]: The good and bad thing with the repo is that we don't have to worry about the new version and security fixes. The downside is obviously that the upgrade could break something.
I've been thinking of using the binary directly to accommodate arm64 but realized it was not heavily used as a feature (yet) and the arm64 user base is small.

Honestly, I just feel lazy to go and upgrade the vault version from time to time... :)

**Comment by leseb** [comment]
Rook will use a different approach, closing this for now.

**Pull request closed without merging**